### DISCUSSION TITLE
Reverse agent connection

### DISCUSSION THREAD

**Discussion by AndreKR** [louislam/dockge#467] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Currently the main installation connects to the other agents, which requires all agents to be reachable from the main installation.

I have some edge hosts that are not reachable at all, but they can reach the main installation via internet.

### ✔️ Solution

There should be a dedicated agent application (or a configuration setting for the normal Dockge application that turns it into an agent) that will connect to the central Dockge installation, receive commands and send data.

### ❓ Alternatives

Of course I could set up Wireguard or Zerotier or something like that in addition to Dockge, but this means extra administrative overhead. Simply connecting via HTTPs to a websocket of the main Dockge installation also means you basically get SSL for free because you probably have already set  it up for the UI.

**Comment by rkroelin** [comment]
I'll second this idea, but for a slightly different use case.  I'd like to be able to deploy the remote dockge (agent) and have it connect to the main dockge installation automatically.  This way you can easily add remote instances and have them show up on your main dockge UI.

**Reply by digitalspaceport** [reply]
This would be nice to see

**Reply by breeegz** [reply]
I actually had this thought as well as I was revamping my homelab.  It would be nice to have an Environment Variable or some config on the remote hosts to connect to the main Dockge Instance.  I found this feature request during my search to see if my idea had any similar requests already posted.

If there was a configuration option for remote connection to the main instance, such as:
```yaml
environment:
  remote_connection: "192.168.1.99:5001" # or FQDN:port
```
Then I could more easily automate the deployment of edge agents without having to point&click the instances together.  I don't see this config option in the docs, so I assume that it doesn't exist.
